### PR TITLE
lnrpc: expose linked htlcs in ListChannels

### PR DESCRIPTION
**Pull request by joostjager**
Fixes https://github.com/lightningnetwork/lnd/issues/4687

### CODE REVIEW

File: lnrpc/rpc.proto
Comment (carlaKC) [inline lnrpc/rpc.proto:1018]: Not sure about the naming on this, what about `forwarding_channel` + `forwarding_htlc_index`? I think that makes sense in the context of an incoming or outgoing htlc, and for our own payments/receives. The word "linked" doesn't have much meaning for an external user (I'd think).

File: htlcswitch/circuit_map.go
Comment (carlaKC) [inline htlcswitch/circuit_map.go:71]: nit: Do these strictly belong in the circiut modifier if they are lookups? maybe just update the comment

**Review by carlaKC — APPROVED** [review]
Just nits ✨

File: rpcserver.go
Comment (carlaKC) [inline rpcserver.go:3589]: This is panicing in itests

**Review by carlaKC — CHANGES_REQUESTED** [review]
On second look, itests are panicing.

File: lnrpc/rpc.proto
Comment (joostjager) [reply lnrpc/rpc.proto:1018]: I was struggling with the name. Will rename to `forwarding_*`

File: rpcserver.go
Comment (joostjager) [reply rpcserver.go:3589]: Hmm, I thought that the itests are also part of the list of github actions in CI. Quickly checked those and they were green. But  apparently need to always check Travis too then.

Comment (carlaKC) [reply rpcserver.go:3589]: They were moved out of actions a while ago because you can't restart an individual action, so one itest flake would mean that every action needs to be run again. While we wait on that feature in actions, it didn't make sense to run them twice.

File: htlcswitch/circuit_map.go
Comment (joostjager) [reply htlcswitch/circuit_map.go:71]: Created new interface

File: rpcserver.go
Comment (joostjager) [reply rpcserver.go]: Ah ok, I missed that. 

Nil check added.

File: rpcserver.go
Comment (carlaKC) [inline rpcserver.go:3603]: s/hop.Source/hop. Source

File: lnrpc/rpc.proto
Comment (carlaKC) [inline lnrpc/rpc.proto:1018]: nit: since we can have a nil outgoing circuit, is it possible that we have a forwarded htlc that will have a 0 forwarding channel (temporarily)? Not the most familiar with that part of the switch, if possible perhaps update the comment so that people don't rely on 0 value meaning own payment.

**Review by carlaKC — APPROVED** [review]
LGTM, going to be very useful to have this info surfaced!

File: rpcserver.go
Comment (joostjager) [reply rpcserver.go:3603]: `hop.Source` is a symbol :)

File: lnrpc/rpc.proto
Comment (joostjager) [reply lnrpc/rpc.proto:1020]: Comment added. I considered making this state explicit, but I don't think it is necessary. Htlcs that need to be forwarded, but haven't even reached the circuit map will also be reported with a zero `forwarding_channel`.

**Review by bhandras — APPROVED** [review]
LGTM :+1:

**Comment by joostjager** [comment]
Anything blocking merge on this one?

**Comment by carlaKC** [comment]
> Anything blocking merge on this one?

No, forgot you don't have merge powers anymore tbh :)

Travis looks confused over the "sample configuration check" - happened to me on a PR that was opened before it was added. Restarting the actions now, then will merge if that does the trick, otherwise you may need to push again to jump it.

**Comment by carlaKC** [comment]
@joostjager won't you push again? Looks like sample config check is stuck.

**Review by cfromknecht — APPROVED** [review]
LGTM 😎

**Pull request merged**
